### ISSUE TITLE
Spreading fragment does not overrides properly

### ISSUE THREAD

**Issue by tomocrafter** [uhyo/nitrogql#42]
### What happened?
It appears that spreading the fragment in query operation does not overrides each fields in that fragment.

### What did you expect?
I expect to this query work:
```graphql
query {
  user(id: 1) {
    id
    ..Description
    ..Name
  }
}

fragment Description on User {
  profile {
    description
  }
}

fragment Name on User {
  profile {
    name
  }
}
```
and it generates type as:
```
{
  // ...
  data: {
    user: {
      id: number,
      profile: {
        description
      }
    }
  }
}
```
instead priotize later fragment:
```
{
  // ...
  data: {
    user: {
      id: number,
      profile: {
        description
      }
    }
  }
}
```

I think this behavior does not sounds right but I'm not sure that fragment should override rather than merging (since GraphQL Code Generator typescript plugin overrides but some other generators will merge, like gqlgen).
I would like nitrogql to override or merge not only the first fragment on query and type.

**Comment by uhyo** [comment]
Thank you for the report! It looks like nitrogql is performing the merge of fragments shallowly. Instead, it should merge deeply. Working on a fix 🙂

**Comment by uhyo** [comment]
Hello, a fix for this has benn released as `1.6.1`. Hope this works for you 🙂
